### DISCUSSION TITLE
How to completely disable video transcoding?

### DISCUSSION THREAD

**Discussion by davidacampos** [immich-app/immich#5126] [Q&A] [answered]
There is an option that says "Don't transcode any videos" which I assume that means for ALL videos, but somehow, even after enabling it, there are some videos getting added to `~/library/encoded_videos" (and this was with a fresh install, in which that setting was set to "no transcode" even before loading any assets).

I tried looking through the code, but I'm not able to find anything in particular.

Any idea on how I can actually fully disable all transcoding operations?

**Comment by jrasm91** [comment, marked as answer]
Besides transcoded videos, Android top motion/live photos require the video be extracted from the original file. Those videos are also stored in that folder.

**Comment by maxexcloo** [comment]
@jrasm91 would those also show up in the TRANSCODE VIDEOS section of Jobs in the admin area?

**Reply by jrasm91** [reply]
What do you mean?

**Reply by maxexcloo** [reply]
The Android/iOS live motion photo conversions, would these come up under transcodes in the Jobs section?

**Reply by jrasm91** [reply]
They are videos yes and the transcoding job does include them, yes.

**Reply by mjulian87** [reply]
@jrasm91 are these transcoded so as not to break playbackm or for another reason?

I would have thought that the expected behavior is that these wouldn't be transcoded under any circumstances if the user has selected 'Don't transcode any videos, may break playback on some clients'.

**Reply by jrasm91** [reply]
Technically the live motion video portion is only _extracted_ from the jpeg and stored in the encoded video folder. For a bunch of reasons that mostly have to do with compatibility with other features within immich, we need the motion part to be its own asset (in the database and on disk), which means it has an "original path" and also an optional "encoded video path". 

Now as a separate asset in the database with a dedicated file on disk it can go through the normal transcoding pipeline like everything else. Turning off transcoding does not prevent the motion video from being extracted, however. Maybe that is where the confusion is coming from. So even with transcoding turned off you will still end up a new asset in the database with it's _original path_ pointing to the encoded-video folder. If transcoding is enabled you would get a _second_ (transcoded) file for this video.

Technically it is possible to stream the video directly out of the image file, since it is just embedded in there, but that becomes quite challenging to integrate with the rest of the asset pipeline in immich.

**Reply by mjulian87** [reply]
Thanks for the explanation 👍
